lib/models/build.coffee

```diff
@@ -40,23 +40,25 @@ getBuildModel = (deps, opts) ->
 	# });
 	#
 	# @example
+	# resin.models.build.getAllByApplication(123).then(function(builds) {
+	#		console.log(builds);
+	# });
+	#
+	# @example
 	# resin.models.build.getAllByApplication('MyApp', function(error, builds) {
 	#		if (error) throw error;
 	#		console.log(builds);
 	# });
 	###
-	exports.getAllByApplication = (name, callback) ->
-		applicationModel().has(name).then (hasApplication) ->
-
-			if not hasApplication
-				throw new errors.ResinApplicationNotFound(name)
+	exports.getAllByApplication = (nameOrId, callback) ->
+		applicationModel().get(nameOrId).then (application) ->
```

Comment:
Yes, exactly. `has` is actually just `get()`ing the application and then suppressing the exact same exception to return a boolean, so we might as well do it more directly.